### PR TITLE
NP Large tensor tests batch 5

### PR DESCRIPTION
**Pull request by Zha0q1**
Batch 5 of large tensor tests.

### CODE REVIEW

**Review by josephevans — APPROVED** [review]

File: tests/nightly/test_np_large_array.py [hunk 1]
```diff
@@ -1312,6 +1312,219 @@ def test_polyval():
...
+    N = 2**16
+    inp1 = np.eye(N)
+    inp1.attach_grad()
+    with mx.autograd.record():
+        out1 = np.trace(inp1)
+        out1.backward()
+    assert out1 == N
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[0, 0] == 1 and inp1.grad[-1, -1] == 1
+    inp2 = np.zeros((2, INT_OVERFLOW))
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:1368]: Can we use inp1 here again so previous memory is released by python ?

File: tests/nightly/test_np_large_array.py [hunk 2]
```diff
@@ -1312,6 +1312,219 @@ def test_polyval():
...
+
+
+@use_np
+def test_tri():
+    N = 2**16
+    data1 = np.tri(N)
+    assert data1.shape == (N, N)
+    assert data1[0, 0] == 1 and data1[-1, -1] == 1
+    assert data1[0, -1] == 0 and data1[-1, 0] == 1
+    data2 = np.tri(2, INT_OVERFLOW, INT_OVERFLOW-2)
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:1386]: same

File: tests/nightly/test_np_large_array.py [hunk 3]
```diff
@@ -1312,6 +1312,219 @@ def test_polyval():
...
+    inp1.attach_grad()
+    with mx.autograd.record():
+        out1 = np.tril(inp1)
+        out1.backward()
+    assert out1.shape == (N, N)
+    assert out1[-1, -1] == 1 and out1[0, -1] == 0 and out1[-1, 0] == 1
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[-1, -1] == 1 and inp1.grad[0, -1] == 0 and \
+        inp1.grad[-1, 0] == 1
+    inp2 = np.ones((2, INT_OVERFLOW))
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:1404]: same

File: tests/nightly/test_np_large_array.py [hunk 4]
```diff
@@ -1312,6 +1312,219 @@ def test_polyval():
...
+    inp1.attach_grad()
+    with mx.autograd.record():
+        out1 = np.triu(inp1)
+        out1.backward()
+    assert out1.shape == (N, N)
+    assert out1[-1, -1] == 1 and out1[0, -1] == 1 and out1[-1, 0] == 0
+    assert inp1.grad.shape == inp1.shape
+    assert inp1.grad[-1, -1] == 1 and inp1.grad[0, -1] == 1 and \
+        inp1.grad[-1, 0] == 0
+    inp2 = np.ones((2, INT_OVERFLOW))
```
Comment (access2rohit) [inline tests/nightly/test_np_large_array.py:1429]: same

**Comment by access2rohit** [comment]
Can you also paste the run log of these tests ? I have left few comments regarding reducing no. of variables to reduce memory used

**Comment by Zha0q1** [comment]
Thanks for reviewing! I just run the tests again locally, sample result:
```

ubuntu@ip-172-31-38-169:~/zhaoqispace$ pytest tests/nightly/test_np_large_array.py::test_triu
=============================================== test session starts ===============================================
platform linux -- Python 3.7.7, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /home/ubuntu/zhaoqispace, inifile: pytest.ini
plugins: remotedata-0.3.2, openfiles-0.4.0, astropy-header-0.1.2, hypothesis-5.8.3, arraydiff-0.3, doctestplus-0.5.0
collected 1 item                                                                                                  

tests/nightly/test_np_large_array.py .                                                                      [100%]

================================================ warnings summary =================================================
tests/nightly/test_np_large_array.py:91
  /home/ubuntu/zhaoqispace/tests/nightly/test_np_large_array.py:91: DeprecationWarning: invalid escape sequence \ 
    '''

tests/nightly/test_np_large_array.py:1534
  /home/ubuntu/zhaoqispace/tests/nightly/test_np_large_array.py:1534: DeprecationWarning: invalid escape sequence \ 
    '''

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================================== 1 passed, 2 warnings in 87.33s (0:01:27) ====================================
```

**Review by access2rohit — APPROVED** [review]

**Comment by Zha0q1** [comment]
@mxnet-bot run ci [centos-gpu, unix-cpu, unix-gpu]

**Comment by Zha0q1** [comment]
Jenkins CI successfully triggered : [unix-cpu, windows-gpu]

**Comment by Zha0q1** [comment]
@mxnet-bot run ci [unix-cpu, windows-gpu]

**Comment by Zha0q1** [comment]
@mxnet-bot run ci [unix_cpu]

**Comment by Zha0q1** [comment]
@mxnet-bot run ci [unix-cpu]

**Pull request merged**